File: android/app/src/main/kotlin/com/edgelink/core/Identity.kt
package com.edgelink.core

data class DeviceIdentity(
    val deviceId: String,
    val name: String,
    val publicKey: ByteArray
) {
    init {
        require(deviceId.matches(Regex("[1-9][0-9]{8}"))) { "Device ID must be 9 digits without a leading zero." }
    }
}

File: mac/Sources/EdgeLinkKit/Identity.swift
import CryptoKit
import Foundation

public struct DeviceIdentity: Sendable {
    public let deviceId: String
    public let name: String
    public let publicKey: Data

    public init(deviceId: String, name: String, publicKey: Data) {
        self.deviceId = deviceId
        self.name = name
        self.publicKey = publicKey
    }
}

public protocol IdentityStore: Sendable {
    func loadIdentity() throws -> DeviceIdentity?
    func saveIdentity(_ identity: DeviceIdentity) throws
}
